### PR TITLE
The solution of the 'credit card' problem

### PR DESCRIPTION
**Pull request by anahittum**

### CODE REVIEW

File: src/main/com/homeworks/week8/credit/card/CreditCard.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package com.homeworks.week8.credit.card;
+
+import java.util.Objects;
+
+public class CreditCard {
+
+    double balance;
```
Comment (gevorgyanarman) [inline src/main/com/homeworks/week8/credit/card/CreditCard.java:7]: should be private

File: src/main/com/homeworks/week8/credit/card/InsufficientBalanceException.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.homeworks.week8.credit.card;
+
+public class InsufficientBalanceException extends RuntimeException {
+
+    public InsufficientBalanceException(String message) {
+
```
Comment (gevorgyanarman) [inline src/main/com/homeworks/week8/credit/card/InsufficientBalanceException.java:6]: the message argument is ignored. please pass it to super class

File: src/test/com/homeworks/week8/credit/card/PaymentServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    @Test
+    public void checkInsufficientBalanceOn4thTry() {
+        CreditCard creditCard = new CreditCard(100, "123456789");
+
+        Assertions.assertThrows(InsufficientBalanceException.class, new Executable() {
+            @Override
+            public void execute() throws Throwable {
+                for (int i = 0; i < 4; i++) {
```
Comment (gevorgyanarman) [inline src/test/com/homeworks/week8/credit/card/PaymentServiceTest.java:27]: here you can not be sure that the exception is thrown right after the 4th invocation.

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

File: src/main/com/homeworks/week8/credit/card/CreditCard.java (hunk 1 above)
Comment (anahittum) [reply src/main/com/homeworks/week8/credit/card/CreditCard.java:7]: Fixed everything but this. Can you please advice how to handle the corresponding changes in pay() method?

File: src/main/com/homeworks/week8/credit/card/PaymentService.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.homeworks.week8.credit.card;
+
+public class PaymentService {
+
+    public void pay(double amountToBePaid, CreditCard creditCard) {
+        if ((creditCard.getBalance() - amountToBePaid) < 0) {
+            throw new InsufficientBalanceException("Not enough balance on the card");
```
Comment (gevorgyanarman) [inline src/main/com/homeworks/week8/credit/card/PaymentService.java:7]: please pass the amount to be paid and balance to the message

File: src/test/com/homeworks/week8/credit/card/PaymentServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        new PaymentService().pay(30, creditCard);
+        new PaymentService().pay(30, creditCard);
+
+        Assertions.assertThrows(InsufficientBalanceException.class, new Executable() {
+
+            @Override
+            public void execute() throws Throwable {
+                new PaymentService().pay(30, creditCard);
+            }
+        });
```
Comment (gevorgyanarman) [inline src/test/com/homeworks/week8/credit/card/PaymentServiceTest.java:33]: please assert the exception message too

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

File: src/main/com/homeworks/week8/credit/card/CreditCard.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+    public double getBalance() {
+        return balance;
+    }
+
+    public String getCardNumber() {
+        return cardNumber;
+    }
+
+    public void debit(double amount) {
```
Comment (gevorgyanarman) [inline src/main/com/homeworks/week8/credit/card/CreditCard.java:24]: let's throw an exception if the amount is bigger than the balance

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

File: src/test/com/homeworks/week8/credit/card/PaymentServiceTest.java (hunk 5 above)
Comment (anahittum) [reply src/test/com/homeworks/week8/credit/card/PaymentServiceTest.java:43]: added a new test for that

**Pull request merged**